Lib/test/test_os.py

```diff
@@ -2148,26 +2148,25 @@ def test_bytes(self):
 @unittest.skipUnless(sys.platform == "win32", "Win32 specific tests")
 @support.skip_unless_symlink
 class Win32SymlinkTests(unittest.TestCase):
-    filelink = 'filelinktest'
-    filelink_target = os.path.abspath(__file__)
-    dirlink = 'dirlinktest'
-    dirlink_target = os.path.dirname(filelink_target)
-    missing_link = 'missing link'
+    dirlink_target = os.path.join(support.TESTFN, 'win32_symlink_tests')
```

Comment:
`support.TESTFN` is just a path, but not an existing directory.

Perhaps add something the following to `setUp()`?
```python3
        tempdir = tempfile.TemporaryDirectory(prefix=support.TESTFN)
        self.addCleanup(tempdir.cleanup)

        self.dirlink_target = os.path.join(tempdir , 'win32_symlink_tests')
        self.filelink_target = os.path.join(self.dirlink_target, support.TESTFN)
        self.dirlink = os.path.join(tempdir , 'dirlinktest')
        self.filelink = os.path.join(tempdir , 'filelinktest')
        self.missing_link = os.path.join(tempdir , 'missinglinktest')
```

With this all of the other existence checks and cleanups may be removed.